src/index.js

```diff
@@ -1,30 +1,96 @@
 import './style.css';
 
-const todos = [
-  {
-    desc: 'todo 1',
-    completed: false,
-    index: 0,
-  },
-  {
-    desc: 'todo 2',
+let todoArr = [];
+
+todoArr = JSON.parse(localStorage.getItem('todos')) || [];
+const form = document.querySelector('form');
+
+// add todo list
+
+const addTodo = (userTask) => {
+  todoArr.push({
+    index: todoArr.length + 1,
+    desc: userTask,
     completed: false,
-    index: 1,
-  },
-  {
-    desc: 'todo 3',
-    completed: true,
-    index: 3,
-  },
-];
-const todoList = document.getElementById('todo-list');
-const ul = document.createElement('ul');
-todos.forEach((todo) => {
-  const li = document.createElement('li');
-  let list = '';
-  list += ` <div class='list'><input type="checkbox"><p>${todo.desc}</p></div> <div class="edit-icon"></div>
-  `;
-  li.innerHTML = list;
-  ul.appendChild(li);
+  });
+  localStorage.setItem('todos', JSON.stringify(todoArr) || []);
+};
+
+// update todo
+
+const updateTask = (value, id) => {
+  const todos = JSON.parse(localStorage.getItem('todos'));
+  const newArr = todos.findIndex((elem) => elem.index === id);
+  todos[newArr].desc = value;
+  localStorage.setItem('todos', JSON.stringify(todos) || []);
+};
```

Comment:
- [ ] [REQUIRED] Kindly ensure you add a function for editing task descriptions. Right now, you have all functions working properly except the edit function which is absent, and the localStorage gets updated. Let's implement this.